python/lsst/meas/extensions/multiprofit/fit_coadd_multiband.py

```diff
@@ -152,142 +206,202 @@ class MultiProFitSourceTask(CatalogSourceFitterABC, fitMB.CoaddMultibandFitSubTa
-        values_init = {
-            g2f.IntegralParameterD: flux,
-            g2f.CentroidXParameterD: cenx,
-            g2f.CentroidYParameterD: ceny,
-            g2f.ReffXParameterD: sig_x,
-            g2f.ReffYParameterD: sig_y,
-            g2f.SigmaXParameterD: sig_x,
-            g2f.SigmaYParameterD: sig_y,
-            g2f.RhoParameterD: -rho,
-        }
-        # Do not initialize PSF size/rho: they'll all stay zero
-        params_psf_init = (g2f.IntegralParameterD, g2f.CentroidXParameterD, g2f.CentroidYParameterD)
-        values_init_psf = {key: values_init[key] for key in params_psf_init}
-        size_major = g2.EllipseMajor(g2.Ellipse(sigma_x=sig_x, sigma_y=sig_y, rho=rho)).r_major
+            # TODO: Add bbox coords or remove
+            cen_x = observation.image.n_cols / 2.0
+            cen_y = observation.image.n_rows / 2.0
+
         # An R_eff larger than the box size is problematic. This should also
         # stop unreasonable size proposals; a log10 transform isn't enough.
         # TODO: Try logit for r_eff?
-        flux_max = 5 * max([np.sum(np.abs(datum.image.data)) for datum in model.data])
-        flux_min = 1 / flux_max
-        limits_flux = g2f.LimitsD(flux_min, flux_max, "unreliable flux limits")
-
-        limits_init = {
-            g2f.IntegralParameterD: limits_flux,
-            g2f.ReffXParameterD: g2f.LimitsD(1e-5, x_max),
-            g2f.ReffYParameterD: g2f.LimitsD(1e-5, y_max),
-            g2f.SigmaXParameterD: g2f.LimitsD(1e-5, x_max),
-            g2f.SigmaYParameterD: g2f.LimitsD(1e-5, y_max),
+        size_major = g2.EllipseMajor(g2.Ellipse(sigma_x=sig_x, sigma_y=sig_y, rho=rho)).r_major
+        limits_size = max(5.0 * size_major, 2.0 * np.hypot(x_max - x_min, y_max - y_min))
+        limits_xy = (1e-5, limits_size)
+        params_limits_init = {
+            g2f.CentroidXParameterD: (cen_x, (x_min, x_max)),
+            g2f.CentroidYParameterD: (cen_y, (y_min, y_max)),
+            g2f.ReffXParameterD: (sig_x, limits_xy),
+            g2f.ReffYParameterD: (sig_y, limits_xy),
+            g2f.SigmaXParameterD: (sig_x, limits_xy),
+            g2f.SigmaYParameterD: (sig_y, limits_xy),
+            g2f.RhoParameterD: (rho, None),
+            # TODO: get guess from configs?
+            g2f.SersicMixComponentIndexParameterD: (1.0, None),
         }
-        limits_init_psf = {g2f.IntegralParameterD: limits_init[g2f.IntegralParameterD]}
-
-        for comp in comps[:n_psfs]:
-            self._init_component(comp, values_init=values_init_psf, limits_init=limits_init_psf)
-        for comp, config_comp in zip(comps[n_psfs:], self.config.sersics.values()):
-            if config_comp.sersicindex.fixed:
-                if g2f.SersicIndexParameterD in values_init:
-                    del values_init[g2f.SersicMixComponentIndexParameterD]
+
+        # TODO: There ought to be a better way to not get the PSF centroids
+        # (those are part of model.data's fixed parameters)
```

Comment:
Agreed. I wonder if adding a flag property to the parameter class would be sufficient.

Reply:
Hmm, what do you mean by that?

To be clear, in this section I'm trying to get fixed centroid parameters (for fixed centroid models) from sources but not the PSF. There is not any way to distinguish between the two, and adding one would be difficult and I think ultimately counterproductive (parameters shouldn't need to know which parametric object(s) they belong to). This is just slightly annoying because you can't get parameters from model.sources because it's a bare list of parametric objects, but I could just shuffle this off to a convenience function in multiprofit, or implement some kind of Sources/SourceList class in gauss2dfit.

Reply:
Honestly I'm not sure. I don't know if I meant to put that comment somewhere else or what, but it doesn't make sense to me either in this context.